dbt/dbt-run.py

```diff
@@ -133,10 +146,113 @@ def run_incremental_models():
   # update incremental models (and tables if there are any)
   subprocess.run(["dbt", "run",  "--profiles-dir", ".dbt", "--exclude", "config.materialized:view"])
 
+def get_pending_doc_count():
+  with connection() as conn:
+      with conn.cursor() as cur:
+          cur.execute(f"""
+              SELECT SUM(pending)
+              FROM {os.getenv('POSTGRES_SCHEMA')}.couchdb_progress
+          """)
+          return cur.fetchone()[0]
+
+def get_batch_ranges():
+  with connection() as conn:
+      with conn.cursor() as cur:
+          cur.execute(f"""
+              SELECT
+                  MIN(saved_timestamp) as start_timestamp,
+                  MAX(saved_timestamp) as end_timestamp
+              FROM {os.getenv('POSTGRES_SCHEMA')}.{os.getenv('POSTGRES_TABLE')}
+          """)
+          result = cur.fetchone()
+          if result is None or len(result) == 0:
+            return []
+          
+          start_timestamp, end_timestamp = result
+          start_timestamp = int(start_timestamp.timestamp())
+          end_timestamp = int(end_timestamp.timestamp())
+          batch_size = int(os.getenv("DBT_BATCH_SIZE") or 10000)
+          return [(start, min(start + batch_size, end_timestamp)) for start in range(start_timestamp, end_timestamp, batch_size)]
+
+def update_batch_status(timestamp, status):
+  with connection() as conn:
+    with conn.cursor() as cur:
+      # insert new entry
+      cur.execute(
+        f"INSERT INTO {os.getenv('POSTGRES_SCHEMA')}.dbt_batch_status (timestamp, status) VALUES (%s, %s);", [timestamp, status]
+      )
+      conn.commit()
+
+def get_last_processed_timestamp():
+  with connection() as conn:
+    with conn.cursor() as cur:
+      cur.execute(f"""
+          SELECT MAX(timestamp)
+          FROM {os.getenv('POSTGRES_SCHEMA')}.dbt_batch_status
+          WHERE status = 'success'
+      """)
+      result = cur.fetchone()
+      if result and result[0]:
+        return result[0]
+      return '1970-01-01 00:00:00'
+    
+def get_max_timestamp():
+  with connection() as conn:
+    with conn.cursor() as cur:
+      cur.execute(f"""
+          SELECT MAX(saved_timestamp)
+          FROM {os.getenv('POSTGRES_SCHEMA')}.document_metadata
+      """)
+      return cur.fetchone()[0]
+   
+def run_dbt_in_batches():
+  print("Running dbt in batches")
+  last_processed_timestamp = get_last_processed_timestamp()
+  batch_size = int(os.getenv("DBT_BATCH_SIZE") or 10000)
+  print(f"Starting new batch with timestamp: {last_processed_timestamp}")
+
+  while True:
+     print(f"Starting new batch with timestamp: {last_processed_timestamp}")
+     update_dbt_deps()
+     result = subprocess.run([
+        "dbt", "run",
+        "--profiles-dir", ".dbt",
+        "--vars", f'{{start_timestamp: "{last_processed_timestamp}", batch_size: {batch_size}}}'
```

Comment:
@njuguna-n I don't see batch_size in the dbt docs https://docs.getdbt.com/docs/ although theres some references to it in the dbt source...is this a new feature in dbt or am I missing the documentation somehow?

Reply:
I added it as a [variable](https://github.com/medic/cht-pipeline/pull/174/files#diff-719858f2cc58b7db36a66f5a66291f4b293b2dccd55d3fcdea202b2196b269feR25) and a change would also be required to the [dbt models](https://github.com/medic/cht-pipeline/pull/174/files#diff-719858f2cc58b7db36a66f5a66291f4b293b2dccd55d3fcdea202b2196b269feR25).

Reply:
So the concern here is that dbt will not select all docs with a certain timestamp? 
Right now, our insert batch size is 1000 (by default), this means that 1000 entries will have the same timestamp. 
What happens when you dbt run over the same document, in case docs with the same timestamp don't make it to the same batch?